geonode-client/templates/geonode-client/map_edit.html

```diff
@@ -0,0 +1,59 @@
+{% load geonode_client_tags %}
+{% load url from future %}
+
+{% client_composer_js	%}
+
+<style>
+	#content {
+		position: relative;
+		margin-top: 15px;
+	}
+	#client-composer-map {
+		height: 100%;
+		width: 100%;
+	}
+</style>
+
+<script type="text/javascript">
+	document.addEventListener('DOMContentLoaded', function () {
+    var defaultConfig = {
+        authStatus: {% if user.is_authenticated %} 200{% else %} 401{% endif %},
+        proxy: '{{ PROXY_URL }}',
+        {% if MAPFISH_PRINT_ENABLED %}
+        printService: "{{GEOSERVER_BASE_URL}}pdf/",
+        {% else %}
+        printService: "",
+        {% endif %}
+        /* The URL to a REST map configuration service.  This service 
+         * provides listing and, with an authenticated user, saving of 
+         * maps on the server for sharing and editing.
+         */
+        rest: "{% url "maps_browse" %}",
+        ajaxLoginUrl: "{% url "account_ajax_login" %}",
+        homeUrl: "{% url "home" %}",
+        portalItems: [{
+            xtype: "container",
+            layout: "fit",
+            height: 81,
+            region: "north"
+        }],
+        localGeoServerBaseUrl: "{{ GEOSERVER_BASE_URL }}",
+        localCSWBaseUrl: "{{ CATALOGUE_BASE_URL }}",
+        csrfToken: "{{ csrf_token }}",
+        tools: [{ptype: "gxp_getfeedfeatureinfo"}],
+    };
+		var config = Object.assign( defaultConfig, {{ config|safe }});
+		var composer = new window.Composer('client-composer-map', "{{SITEURL}}");
+		composer.mapConfig = config;
+		composer.mapId = "{{mapId}}";
```

Comment:
maybe we have only one template for edit and new. the only change is the set mapId. 
We could test either if it is empty than not set it, or change the internals to test if it is empty instead of undefined.